### ISSUE TITLE
fig.suptitle overlapping with axis using constrained_layout

### ISSUE THREAD

**Issue by asteppke** [matplotlib/matplotlib#13672]
I want to add a title to a matplotlib figure. Using the newer `constrained_layout` the title is nicely aligned with the axis. 

Unfortunately under certain circumstances the layout still overlaps as shown in the following examples in the Jupyter notebook (with different backends):

```python
%matplotlib notebook 
import numpy as np
from matplotlib import pyplot as plt

# example with low height
fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(6, 3), constrained_layout=True)
axes[0,0].plot([0,1,2], label='test')
axes[1,0].plot([0,1,2], label='test 2')
fig.suptitle('a longer title') 
```

This results in
![image](https://user-images.githubusercontent.com/5851636/54355839-6c54ab80-465a-11e9-8c29-ecc19e7cebf7.png)

But now increasing the plot height using the modified example
    
```python
[...]
fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(6, 12), constrained_layout=True)
[...]
```

shows the following overlapping title:

![image](https://user-images.githubusercontent.com/5851636/54355911-9ad28680-465a-11e9-89b8-0c61cf9fc6d2.png)

Is there anything that can be done here to find out why this is not taking into account the larger plot dimensions?

_Originally posted by @asteppke in https://github.com/matplotlib/matplotlib/issues/5355#issuecomment-472828628_

**Comment by jklymak** [comment]
Thanks @asteppke   I'll try and look at this in the next little while. I think everything is "working" but "working" isn't quite what we want.

**Comment by jklymak** [comment]
So, on my machine, this is caused by the Qt5 window not being allowed to be larger than the size of the screen, so the mismatch between the requested size and the actual size causes a positioning mismatch.  If you save a png *before* you try to show the plot, the layout works fine.  

Not sure what we can do about this...

![Booo](https://user-images.githubusercontent.com/1562854/54387842-a2ddf680-4659-11e9-91a6-cec0495bef2e.png)

**Comment by asteppke** [comment]
Thank you for looking into this. I only tested this in the Jupyter notebook so far and there it is reproducible with several backends (TkAgg, NbAgg), so at least it is not tied to Qt5 or a specific windowing framework.

**Comment by jjnurminen** [comment]
FWIW, while waiting for the real fix, I added '\n' at the end of my suptitle. This seems to provide the perfect amount of whitespace :)

**Comment by jklymak** [comment]
From #14814, a more reproducible error from @ArchangeGabriel

```python
import matplotlib.pyplot as plt
plt.rcParams = plt.rcParamsDefault
plt.rcParams['figure.constrained_layout.use'] = True
plt.rcParams['figure.constrained_layout.h_pad'] = 0.01
plt.title('Test with b')
plt.suptitle('Test with p')
```

**Comment by ArchangeGabriel** [comment]
@jjnurminen In my case, this adds way too much white space, because the font is big w.r.t. to the plot size.

**Comment by ArchangeGabriel** [comment]
@jklymak Awesome that you could get this fixed in time for the 3.1.0 release! Thanks a lot. :)
